packages/desktop-client/src/components/autocomplete/Autocomplete.tsx

```diff
@@ -302,6 +302,27 @@ function SingleAutocomplete<T extends Item>({
 
   const filtered = isChanged ? filteredSuggestions || suggestions : suggestions;
 
+  function customSort(obj, value) {
+    const name = obj.name.toLowerCase();
+    const groupName = obj.group ? obj.group.name.toLowerCase() : '';
+    value = value.toLowerCase();
+
+    if (obj.id === 'split') {
+      return -2;
+    }
+    if (name.includes(value)) {
+      return -1;
+    }
+    if (groupName.includes(value)) {
+      return 0;
+    }
+    return 1;
+  }
+
+  if (isChanged) {
+    filtered.sort((a, b) => customSort(a, value) - customSort(b, value));
+  }
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏a couple of notes on the sorting logic:

1. the generic autocomplete component is not a good place for the custom sorting logic that is applicable only to category autocomplete - a better location would be in `CategoryAutocomplete`
2. instead of using `.sort` (which mutates the array in-place thus causing extra re-renders in the component tree) -  use `.toSorted`
3. remember to `useMemo` the result - a small change to improve the perf.
4. missing TS types for the custom sort util; plus - if it's a generic - it should be moved _outside_ the component - thus reducing the amount of re-renders